src/pamiq_core/model/container.py

```diff
@@ -8,7 +8,7 @@ class InferenceModelsDict(UserDict[str, InferenceModel]):
     """Wrapper class for models to infer."""
 
 
-class TrainingModelsDict(UserDict[str, TrainingModel]):
+class TrainingModelsDict(UserDict[str, TrainingModel[Any]]):
```

Comment:
確かに`Any`でも型チェック通りはするんだけど、可読性重視してこっちにも`bound=InferenceModel`欲しいかも
```suggestion
T = TypeVar('T', bound=InferenceModel)
class TrainingModelsDict(UserDict[str, TrainingModel[T]]):
```

Reply:
@zassou65535 `Any`型には意図的にしているんだけど、説明が難しいんだよな。型変数周りは。

Reply:
@zassou65535 本当にそれをやっても大丈夫か、ちゃんと自分で考えてみて。その上でコメントしてほしいな
https://kdotdev.com/kdotdev/python-covariance-contravariance-variance

Reply:
ちなみに型変数を導入すると、 全て同じ `InferenceModel`型を持つ `TrainingModel`しか受け付けなくなる。